.github/workflows/test-integration.yml

```diff
@@ -20,14 +21,21 @@ jobs:
       - uses: actions/setup-go@v3
         with:
           go-version: 1.18.x
-      - name: Install dependencies
-        run: |
-          if [ "$RUNNER_OS" == "Linux" ]; then
-            sudo apt update && sudo apt install build-essential
-          fi
-        shell: bash
+      - name: Install dependencies on Linux
+        if: runner.os == 'Linux'
+        run: sudo apt update && sudo apt install build-essential
+      - name: Install dependencies on Windows
+        if: runner.os == 'Windows'
+        run: choco upgrade mingw cmake -y --no-progress
+
       - uses: actions/cache@v3
         with:
           path: ~/go/pkg/mod
           key: ${{ matrix.os }}-go-${{ hashFiles('**/go.sum') }}
-      - run: make test-integration
+
+      - name: test-integration on Unix
+        if: runner.os != 'Windows'
+        run: make test-integration
+      - name: test-integration on Windows
+        if: runner.os == 'Windows'
```

Comment:
maybe easier to create dedicated .yml file for win - because `if` in every line (but overall .yml looks simple enough).